tests/unit/test_webhooks.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Tests for our webhooks."""
+
+import json
+import os
+import unittest
+
+import flask
+
+from ochazuke import create_app
+from ochazuke.models import db
+from ochazuke.models import Issue
+from ochazuke.models import Event
+from ochazuke.webhooks import helpers
+
+
+# The key is for testing and computing the signature.
+# TODO: figure out storage/retrieval configuration
+key = 'SECRETS'
+
+
+# Some machinery for opening our test files
+def event_data(filename):
+    """Return a tuple with the content and its signature."""
+    current_root = os.path.realpath(os.curdir)
+    events_path = 'tests/fixtures/webhooks'
+    path = os.path.join(current_root, events_path, filename)
+    with open(path, 'r') as f:
+        json_event = json.dumps(json.load(f))
+    signature = 'sha1={sig}'.format(
+        sig=helpers.get_payload_signature(key, json_event))
+    return json_event, signature
+
+
+class TestWebhooks(unittest.TestCase):
+    """Tests for our webhooks handler."""
+
+    def setUp(self):
+        """Set up tests."""
+        self.app = create_app(test_config={})
+        self.client = self.app.test_client()
+        # binds app to the current context
```

Comment:
@karlcow This is where I'm going wrong with the app context. This code came from multiple sources, so it's probably not the right combination. When I was testing in the REPL, this seemed to work:
```
app = create_app()
app.app_context().push()
```